main.cpp

```diff
@@ -34,61 +34,42 @@ struct F                                //4
 {
     T* compare(T* a, T* b) //5
     {
-        if (a != nullptr && b != nullptr)
-        {
-            if( a->value < b->value ) return a;
-            if( a->value > b->value ) return b;
-        }
+        if( a->value < b->value ) return a;
+        if( a->value > b->value ) return b;
```

Comment:
While you cannot change the return type, you can change the function arguments to be references
- [ ] revise